tests/qnodes/test_qnode_decorator.py

```diff
@@ -140,6 +140,18 @@ def circuit(a):
     circuit.h = new_step_size
     assert circuit.h == new_step_size
 
+
+def test_reversible_diff_method():
+    """Test that a ReversibleQNode can be created via the qnode decorator"""
+    dev = qml.device('default.qubit', wires=1)
+
+    @qnode(dev, diff_method="reversible")
+    def circuit(a):
+        qml.RX(a, wires=0)
+        return qml.expval(qml.PauliZ(wires=0))
+
+    assert isinstance(circuit, ReversibleQNode)
```

Comment:
Would be good to also test validation here once implemented (e.g., attempting to use reversible mode with a non-QubitDevice device, or one that doesn't have `dev._pre_rotated_state`).